src/transfers-history/services/SpokePoolEventsQueryService.ts

```diff
@@ -0,0 +1,98 @@
+import { TypedEvent } from "@across-protocol/contracts-v2/dist/typechain/common";
+import { FundsDepositedEvent, FilledRelayEvent } from "@across-protocol/contracts-v2/dist/typechain/SpokePool";
+import { BigNumber, providers } from "ethers";
+
+import { TransfersRepository } from "../adapters/db/transfers-repository";
+import { Logger } from "../adapters/logger";
+import { ISpokePoolContractEventsQuerier } from "../adapters/web3";
+import { ChainId } from "../adapters/web3/model";
+import { clientConfig } from "../config";
+import { Transfer } from "../model";
+
+export class SpokePoolEventsQueryService {
+  private latestBlockNumber: number | undefined;
+
+  constructor(
+    private chainId: ChainId,
+    private provider: providers.Provider,
+    private eventsQuerier: ISpokePoolContractEventsQuerier,
+    private logger: Logger,
+    private transfersRepository: TransfersRepository,
+    private depositorAddr?: string
```

Comment:
i feel like you should not make more than one of these per depositor address, othewrise you will re-query a lot of data you already have

Reply:
the `queryFilter` calls contains the depositor address as a filter, so I don't think I query data I already have